src/BatteryStateOfChargeService.cpp

```diff
@@ -16,21 +19,63 @@ BatteryStateOfChargeService::~BatteryStateOfChargeService()
 
 double BatteryStateOfChargeService::totalAmpHoursUsed() const
 {
-    return 0.0;
+    return BATTERY_AMP_HOUR_CAPACITY-AmpHours;
 }
 
 bool BatteryStateOfChargeService::isCharging() const
 {
-    return false;
+    if (Current < 0)
+    {
+        return false;
+    }
+    else
+    {
+        return true;
+    }
 }
 
 QTime BatteryStateOfChargeService::timeWhenChargedOrDepleted() const
 {
-    return QTime::currentTime();
+    double timeRemaining = (BATTERY_AMP_HOUR_CAPACITY-AmpHours)/inCurrent;
+    timeRemaining = qAbs(timeRemaining)/CONVERT_TO_MILLISECONDS;
+    return QTime(0, 0, 0, 0).addMSecs(timeRemaining);
 }
 
 void BatteryStateOfChargeService::addData(const BatteryData& batteryData)
 {
-    Q_UNUSED(batteryData);
-    // Update your variables here.
+
+    QTime currentTime = batteryData.time;
+    changeTime = abs(initialTime.msecsTo(currentTime));
+    changeTime = changeTime * CONVERT_TO_MILLISECONDS;
+    initialTime = currentTime;
+    if ((Current > 0 && Current < 10000000000) || Current < 0 )
```

Comment:
What's the reasoning behind this if statement? I'm not really understanding where these boundaries are coming from.